src/graph/data_apis.cc

```diff
@@ -0,0 +1,61 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file graph/graph.cc
+ * \brief DGL graph index APIs
+ */
+
+#include <string.h>
+#include <dgl/graph.h>
+#include "../c_api_common.h"
+
+using dgl::runtime::DGLArgs;
+using dgl::runtime::DGLArgValue;
+using dgl::runtime::DGLRetValue;
+using dgl::runtime::PackedFunc;
+using dgl::runtime::NDArray;
+
+namespace dgl {
+
+std::unordered_map<std::string, DLDataType> dtype_map = {
+  {"float32", DLDataType{kDLFloat, 32, 1}},
+  {"int", DLDataType{kDLInt, 32, 1}},
+  {"uint", DLDataType{kDLUInt, 32, 1}},
+};
+
+DGL_REGISTER_GLOBAL("contrib.graph_store._CAPI_DGLCreateSharedMem")
```

Comment:
`DGLCreateSharedMem` -> `DGLOpenSharedMem`

Reply:
why use `open`? it is creating/allocating shared memory

Reply:
Because if `is_create` is False then you are just opening an existing one?

Reply:
i rename it to _CAPI_DGLAllocateSharedMem.